src/herring/tl_helpers.hpp

```diff
@@ -0,0 +1,305 @@
+  simple_model<float, std::uint16_t, std::uint16_t, std::uint32_t, std::vector<float>>,
+
+  simple_model<float, std::uint16_t, std::uint32_t, std::uint32_t, std::uint32_t>,
+  simple_model<float, std::uint16_t, std::uint32_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<float, std::uint16_t, std::uint32_t, std::uint32_t, float>,
+  simple_model<float, std::uint16_t, std::uint32_t, std::uint32_t, std::vector<float>>,
+
+  simple_model<float, std::uint32_t, std::uint16_t, std::uint32_t, std::uint32_t>,
+  simple_model<float, std::uint32_t, std::uint16_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<float, std::uint32_t, std::uint16_t, std::uint32_t, float>,
+  simple_model<float, std::uint32_t, std::uint16_t, std::uint32_t, std::vector<float>>,
+
+  simple_model<float, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t>,
+  simple_model<float, std::uint32_t, std::uint32_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<float, std::uint32_t, std::uint32_t, std::uint32_t, float>,
+  simple_model<float, std::uint32_t, std::uint32_t, std::uint32_t, std::vector<float>>,
+
+  simple_model<double, std::uint16_t, std::uint16_t, std::uint32_t, std::uint32_t>,
+  simple_model<double, std::uint16_t, std::uint16_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<double, std::uint16_t, std::uint16_t, std::uint32_t, double>,
+  simple_model<double, std::uint16_t, std::uint16_t, std::uint32_t, std::vector<double>>,
+
+  simple_model<double, std::uint16_t, std::uint32_t, std::uint32_t, std::uint32_t>,
+  simple_model<double, std::uint16_t, std::uint32_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<double, std::uint16_t, std::uint32_t, std::uint32_t, double>,
+  simple_model<double, std::uint16_t, std::uint32_t, std::uint32_t, std::vector<double>>,
+
+  simple_model<double, std::uint32_t, std::uint16_t, std::uint32_t, std::uint32_t>,
+  simple_model<double, std::uint32_t, std::uint16_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<double, std::uint32_t, std::uint16_t, std::uint32_t, double>,
+  simple_model<double, std::uint32_t, std::uint16_t, std::uint32_t, std::vector<double>>,
+
+  simple_model<double, std::uint32_t, std::uint32_t, std::uint32_t, std::uint32_t>,
+  simple_model<double, std::uint32_t, std::uint32_t, std::uint32_t, std::vector<std::uint32_t>>,
+  simple_model<double, std::uint32_t, std::uint32_t, std::uint32_t, double>,
+  simple_model<double, std::uint32_t, std::uint32_t, std::uint32_t, std::vector<double>>
+>;
+
+template<std::size_t model_variant_index, typename tl_threshold_t, typename tl_output_t>
+auto convert_dispatched_model(treelite::ModelImpl<tl_threshold_t, tl_output_t> const& tl_model) {
+  using model_type = std::variant_alternative_t<model_variant_index, tl_dispatched_model>;
+  auto result = model_type{};
+  result.use_inclusive_threshold = false;
+
+  result.trees.reserve(tl_model.trees.size());
+  std::transform(
```

Comment:
Yes, but it's not as easy as throwing a `std::execution::par` at it. The order of the trees can matter. If it's all right, I'd like to defer optimization of the actual model conversion to a later PR since we don't care too much about performance for that step in Triton and there's probably a lot of low-hanging fruit for improving it that we can tackle later on. Just created a follow-up issue here: https://github.com/triton-inference-server/fil_backend/issues/208